src/cmd/remove.nim

```diff
@@ -27,10 +27,31 @@ proc paxRemove*(name: string): void =
   echoMod(cfMod, moreInfo = true)
   echo ""
 
-  returnIfNot promptYN("Are you sure you want to remove this mod?", default = true)
-  
+  returnIfNot promptYN("Are you sure you want to remove this mod?",
+      default = true)
+
   echoInfo "Removing ", cfMod.name.cyanFg, ".."
   manifest.removeMod(cfMod.projectId)
 
+  let cfModFiles = waitFor(fetchModFiles(cfMod.projectId))
+  let selectedCfModFile = cfModFiles.selectModFile(manifest, strategy)
+  if not selectedCfModFile.isNone:
+    let cfModFile = selectedCfModFile.get()
+
+    if len(cfModFile.dependencies) > 0:
+      if promptYN("Do you want to remove dependencies?", default = true):
+        for id in cfModFile.dependencies:
+          let cfMod = waitFor(fetchMod(id))
+          if not manifest.isInstalled(id):
+            continue
+
+          let cfModFiles = waitFor(fetchModFiles(id))
+          let selectedCfModFile = cfModFiles.selectModFile(manifest, strategy)
+          if selectedCfModFile.isNone:
+            echoError "Warning: unable to resolve dependencies."
+          let cfModFile = selectedCfModFile.get()
+          echoInfo "Removing ", cfMod.name.cyanFg, ".."
+          manifest.removeMod(id)
+
```

Comment:
Ok, adding a dependency looks very good (thanks for your contribution again ;) ), but removing a dependency is a bit more difficult.

What if the user first installs `JEI`, then installs `Just Enough Resources`. then removes `Just Enough Resources`? With your code, `JEI` would get removed as well although the user probably didn't intend to do so.
What if the user installs `Yung's Better Strongholds` (depends on `Yungs API`), then adds `Yung's Better Mineshafts` (depends on `Yungs API`), then removes `Yung's Better Strongholds`? `Yungs API` would be removed although it still is required by `Yung's Better Mineshafts`.


I think the best idea would be to track
- the dependency projectIDs of a file: (`dependencies: [123, 234]`)
- and whether the file was installed explicitly (with `./pax add`) or not (as a dependency): (`standalone: true`)
in the `manifest.json` file, like so:

```json5
{
  ...
  "files": [
    {
      "projectID": 238603,
      "fileID": 3156163,
      "required": true,
      "__paxMeta": {
        "dependencies": [123, 234],
        "standalone": true
      }
    },
    ...
  ]
}
```

This requires some changes though:
- When installing mods:
  - write the installed mod with the dependency projectIDs & `"standalone": true` to the manifest like outlined above
  - write dependencies of the installed mod with `"standalone": false` to the manifest (if they were not already installed).
- When removing mods, you'll need to:
  - check if the mod to be removed is a dependency of another mod (iterate over the manifest files)
    - if yes, abort, otherwise continue
  - save the `dependencies` array of the selected mod which is to be removed
  - remove the selected mod
  - remove all files that have a projectID in the saved `dependencies` array ONLY if they are set to `"standalone": false`

I *think* that should solve all the aforementioned issues. Don't be afraid to come forward if you think I missed something.